keps/sig-scheduling/5229-asynchronous-api-calls-during-scheduling/README.md

```diff
@@ -0,0 +1,998 @@
+Some delay might potentially appear due to side effects, especially when there will be problems with the kube-apiserver,
+so some higher-level mechanism such as (1.1) or (1.2) would need to prevent pod update races.
+
+Pros:
+- Simple to implement if the appropriate race handling is chosen.
+- Can easily be extended with a simple queue and worker pool to limit number of goroutines.
+
+Cons:
+- Does not inherently support delaying calls.
+- Higher-level mechanisms (like 1.1 or 1.2) would be needed to prevent pod update races.
+- `nominatedNodeName` scenario support would require more effort in (1.1) or (1.2).
+
+
+#### 2.2: Make the API calls queued
+
+To make asynchronous dispatching more advanced, a queueing approach could be explored.
+A queue might understand what the API calls are intended to do and eventually delay, skip, or merge them,
+e.g., don't set `nominatedNodeName` when pod binding is enqueued.
+Initially, it could be a framework, which might be extended in the future, e.g., by introducing the possibility of setting delays.
+
+However, it is questionable what should happen if two update API calls for the same pod are enqueued.
+This might not happen in (1.1) and (1.2) if we wait for the previous status update call to complete or terminate it.
+Otherwise, as currently the update is done on a copy of a pod, these two might collide. If the update were to be done on the original pod object,
+it might be possible to simply decide what API calls should be applied for a pod:
+- Status update (patch): Apply newest API call
+- Binding: Ignore status update API calls
+- Delete (in preemption): Ignore status update as well as binding API calls
+
+```go
+type APICallType string
+
+const (
+  StatusUpdate APICallType = "status_update"
+  Binding      APICallType = "binding"
+  Delete       APICallType = "delete"
+)
+
+type PodAPICall struct {
+	podID    types.UID
+	callType APICallType
+	fn       func()
+}
+
+type APIQueue struct {
+  ...
+}
+
+func (aq *APIQueue) Add(podAPICall PodAPICall) {
+  // If API call for specific podID is already enqueued,
+  // check the callType and skip or replace the call depending on precedence.
+  ...
+}
+
+func (aq *APIQueue) Run() {
+  // Dispatch limited number of goroutines if queue is non empty.
+  ...
+}
+```
+
+Pros:
+- Allows for advanced goroutine dispatching logic.
+- Can potentially delay, skip, or merge API calls based on type (e.g., skip `nominatedNodeName` if binding is pending).
+- All collisions could be resolved at the queue level, not relying on higher-level mechanisms (like 1.1 or 1.2).
+- Allows for (1.3) where all scenarios can be supported without additional structures.
+- Provides a framework that can be extended in the future.
+
+Cons:
+- Requires complex logic to handle potential conflicts between different update types for the same pod.
+- Needs a clear strategy for how to update the in-memory pod object during scheduling.
+
+
+#### 2.3: Send API calls through a kube-scheduler's cache
```

Comment:
By the way, actually this cache idea looks like also part of `1: Where and how to handle API calls in the kube-scheduler` as well? 
Maybe it's not clear what to talk in 1 and what to talk in 2. Because, as we see, we're describing the queue thingy twice in both sections.